File: musicgen.py
from audiocraft.models import MusicGen
import io

async def generate_music(prompt: str, duration: int, model_type: str):
    model = MusicGen.get_pretrained(model_type)
    model.set_generation_params(duration=duration)
    output = model.generate([prompt])
    buffer = io.BytesIO()
    output[0].save(buffer, format="wav")
    buffer.seek(0)
    return buffer

File: musicgen.js
export async function generateMusic({ prompt, duration, model }) {
  const response = await fetch('http://localhost:8000/api/generate-music', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ prompt, duration, model_type: model }),
  });
  if (!response.ok) throw new Error('Error generando música');
  return await response.blob();
}